src/clr-angular/popover/dropdown/dropdown-trigger.spec.ts

```diff
@@ -49,5 +49,23 @@ export default function(): void {
     it('declares itself to the DropdownFocusHandler', function(this: Context) {
       expect(this.getClarityProvider(DropdownFocusHandler).trigger).toBe(this.clarityElement);
     });
+
+    it('default value should be added when inputs are not receiving data', function(this: Context) {
+      expect(this.clarityDirective.clrAriaLabelDropdownOpen).not.toBeUndefined();
+      expect(this.clarityDirective.clrAriaLabelDropdownClose).not.toBeUndefined();
+    });
+
+    it('value received through inputs should be set to aria-label', function(this: Context) {
+      const ifOpenService = this.getProvider(IfOpenService);
+      ifOpenService.open = false;
+      this.clarityDirective.clrAriaLabelDropdownOpen = 'New value for open';
+      this.clarityElement.setAttribute('aria-label', this.clarityDirective.setAriaLabel);
```

Comment:
```
      this.clarityElement.setAttribute('aria-label', this.clarityDirective.setAriaLabel);
```
This is the thing that you want to test, but you do it instead of the component. Better will be to just set the `Input`  and let the component do the job, by setting the correct value for `aria-label` and pass whatever flows it needs to do. For example, `setAriaLabel` could have additional logic that may or may not change the value.

```
@Component({
  template: `
    <button clrDropdownTrigger>Hello world</button>
  `,
  // These services are declared here because they need the renderer
  providers: [FocusService, DROPDOWN_FOCUS_HANDLER_PROVIDER],
})
class SimpleTest {}
```

Create or update the above template to include `clrAriaLabelDropdownClose` and `clrAriaLabelDropdownOpen` as inputs and change them to see if they apply the correct behavior. There are a lot of examples out there.

Reply:
i ll keep the template as it is because i need it without inputs for the first test. i ll just set the new values of the inputs and use detectchanges() for the second.
```const ifOpenService = this.getProvider(IfOpenService);
      ifOpenService.open = false;
      this.clarityDirective.clrAriaLabelDropdownOpen = 'New value for open';
      this.detectChanges();
      expect(this.clarityElement.getAttribute('aria-label')).toBe('New value for open');
```
How s this way? It is ok?

Reply:
By default, the component should set the default value. And you could change attributes this way.

```js
@Component({
  template: `
    <button clrDropdownTrigger [clrAriaLabelDropdownOpen]="openLabel" [clrAriaLabelDropdownClose]="closeLabel" >Hello world</button>
  `,
  // These services are declared here because they need the renderer
  providers: [FocusService, DROPDOWN_FOCUS_HANDLER_PROVIDER],
})
class SimpleTest {
  openLabel: string
  closeLabel: string
}
```

Inside the test you could update and change the value:

```js
expect(this.clarityElement.getAttribute('aria-label')).toBe('default text when no value is passed');
this.testComponent.openLabel = "some text";
this.detectChanges();
expect(this.clarityElement.getAttribute('aria-label')).toBe('some text');
```